### PR TITLE
serverless beauty

### PR DESCRIPTION
**Pull request by nielsmijnster**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ module.exports = {
         // necessary to make import rules to find files
     'import/resolver': {
       node: {
-        extensions: ['.js', '.jsx', '.ts', '.tsx'],
+        extensions: ['.js', '.jsx', '.src/functions/handler.ts', '.tsx'],
```
Comment (flaviostutz) [inline .eslintrc.js]: Why do you need this specific file indication?

File: .serverless/cloudformation-template-create-stack.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
```
Comment (flaviostutz) [inline .serverless/cloudformation-template-create-stack.json]: This file is auto generated and shouldn't be committed.
The entire .serverless folder shouldn't be committed.

File: package.json [hunk 3]
```diff
@@ -25,17 +38,29 @@
     "posttest": "npm run lint",
     "slug": "ts-node slug.ts",
     "test": "jest",
-    "cow-slug": "node dist/cow-slug.js",
+    "cow-slug": "node lib/cow-slug.js",
```
Comment (flaviostutz) [inline package.json]: Normally we use à dir called "dist" for autogenerated files. It's better than "lib"

File: serverless.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+custom:
+  esbuild:
+    bundle: true
+    minify: false
+    sourcemap: true
+    exclude: 'aws-sdk'
+    target: node14
+    define:
+      'require.resolve': undefined
```
Comment (flaviostutz) [inline serverless.yml]: What does this line does?

File: tsconfig.json [hunk 5]
```diff
@@ -1,34 +1,45 @@
...
+        "removeComments": true,
+        "sourceMap": true,
+        "outDir": "lib"
   },
   "parserOptions.projects": {
     "include": [
       "src/**/*"
     ],
   },
+  "include": ["src/**/*.ts", "serverless.ts", "index.ts", "index.ts", "src/functions/handler.ts", "src/functions/handler.js"],
```
Comment (flaviostutz) [inline tsconfig.json]: Why did you include these specific rules instead of the more general rules that were present before?

File: tsconfig.json [hunk 6]
```diff
@@ -1,34 +1,45 @@
...
+  "exclude": [
+    "node_modules/**/*",
```
Comment (flaviostutz) [inline tsconfig.json:30]: If you indicate only the dir name is already enough.

File: tsconfig.paths.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@functions/*": ["src/functions/*"],
+      "@libs/*": ["src/libs/*"]
```
Comment (flaviostutz) [inline tsconfig.paths.json]: Why do you need to add "libs" here?

**Review by flaviostutz — CHANGES_REQUESTED** [review]
Nice work! I requested some small improvements to make things simpler.

File: .eslintrc.js (hunk 1 above)
Comment (nielsmijnster) [reply .eslintrc.js]: I forgot to change it back, one time it did not work and then I tried this, but changed now!

File: .serverless/cloudformation-template-create-stack.json (hunk 2 above)
Comment (nielsmijnster) [reply .serverless/cloudformation-template-create-stack.json]: removed

Comment (nielsmijnster) [reply .serverless/cloudformation-template-create-stack.json]: I mean, put it in gitignore, or does it stay in the repo then?

File: package.json (hunk 3 above)
Comment (nielsmijnster) [reply package.json]: haha we removed the dist together and then I saw somewhere this autogenerated lib and I thought, lets call it like that. But if dist is better we go for that.
I changed it back now:)

File: serverless.yml (hunk 4 above)
Comment (nielsmijnster) [reply serverless.yml]: nothing, i realize now that I could be more meticulous regarding all the code. I removed this lines.

File: tsconfig.json (hunk 5 above)
Comment (nielsmijnster) [reply tsconfig.json]: this was from my problems before I had it working, changed it

File: tsconfig.json (hunk 6 above)
Comment (nielsmijnster) [reply tsconfig.json:30]: yes I did that.

File: tsconfig.paths.json (hunk 7 above)
Comment (nielsmijnster) [reply tsconfig.paths.json]: I was not aware of this, removed now!

File: .serverless/cloudformation-template-update-stack.json [hunk 8]
```diff
@@ -167,7 +167,7 @@
           "S3Bucket": {
             "Ref": "ServerlessDeploymentBucket"
           },
-          "S3Key": "serverless/full-lambda-function/dev/1666098617776-2022-10-18T13:10:17.776Z/full-lambda-function.zip"
+          "S3Key": "serverless/full-lambda-function/dev/1666255668655-2022-10-20T08:47:48.655Z/full-lambda-function.zip"
```
Comment (flaviostutz) [inline .serverless/cloudformation-template-update-stack.json]: The entire .serverless folder should be deleted and part of gitignore then

File: tsconfig.paths.json [hunk 9]
```diff
@@ -2,8 +2,7 @@
   "compilerOptions": {
     "baseUrl": ".",
     "paths": {
-      "@functions/*": ["src/functions/*"],
-      "@libs/*": ["src/libs/*"]
+      "@functions/*": ["src/functions/*"]
```
Comment (flaviostutz) [inline tsconfig.paths.json:5]: Is this file needed?

**Review by flaviostutz — CHANGES_REQUESTED** [review]
Pls check my comments

File: tsconfig.paths.json (hunk 9 above)
Comment (nielsmijnster) [reply tsconfig.paths.json:5]: yes, the tsconfig.json needs this route, otherwise I got an error during the build process of the cow-slug stuff.

File: .serverless/cloudformation-template-update-stack.json (hunk 8 above)
Comment (nielsmijnster) [reply .serverless/cloudformation-template-update-stack.json]: deleted now

**Review by flaviostutz — APPROVED** [review]
